x-pack/plugins/fleet/server/services/secrets.ts

```diff
@@ -759,3 +765,28 @@ function _getInputSecretVarDefsByPolicyTemplateAndType(packageInfo: PackageInfo)
     {}
   );
 }
+
+/**
+ * Given an array of secret paths, existing secrets, and a package policy, generates a
+ * new package policy object that includes resolved secret reference values at each
+ * provided path.
+ */
+function getPolicyWithSecretReferences(
+  secretPaths: SecretPath[],
+  secrets: Secret[],
+  packagePolicy: NewPackagePolicy
+) {
+  const result = JSON.parse(JSON.stringify(packagePolicy));
+
+  secretPaths.forEach((secretPath, i) => {
+    secretPath.path.reduce((acc, val, j) => {
+      if (j === [...secretPath.path].length - 1) {
+        acc[val].value = toVarSecretRef(secrets[i].id);
+      }
+
+      return acc[val];
```

Comment:
it is possible that `acc[val]` is not defined? in that case should we do something like `acc[val] = {}; return acc[val];` (this mimic what lodash set is doing)